cli/src/main/java/com/box/l10n/mojito/cli/command/checks/ControlCharacterChecker.java

```diff
@@ -0,0 +1,64 @@
+package com.box.l10n.mojito.cli.command.checks;
+
+import com.box.l10n.mojito.cli.command.extraction.AssetExtractionDiff;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.box.l10n.mojito.cli.command.extractioncheck.ExtractionCheckNotificationSender.QUOTE_MARKER;
+
+public class ControlCharacterChecker extends AbstractCliChecker {
+
+    class ControlCharacterCheckerResult {
+        boolean isSuccessful = true;
+        String failureText = "";
+    }
+
+    @Override
+    public CliCheckResult run(List<AssetExtractionDiff> assetExtractionDiffs) {
+        List<String> failures = getSourceStringsFromDiff(assetExtractionDiffs).stream()
+                .map(this::getControlCharacterCheckerResult)
+                .filter(result -> !result.isSuccessful)
+                .map(result -> result.failureText)
+                .collect(Collectors.toList());
+        CliCheckResult result = createCliCheckerResult();
+        if (!failures.isEmpty()) {
+            result.setSuccessful(false);
+            result.setNotificationText(getNotificationText(failures));
+        }
+        return result;
+    }
+
+    private ControlCharacterCheckerResult getControlCharacterCheckerResult(String source) {
+        ControlCharacterCheckerResult result = new ControlCharacterCheckerResult();
+        char[] characters = source.toCharArray();
+        List<Integer> indexMatches = new ArrayList<>();
+        for (int i = 0; i < characters.length; i++) {
+            if (Character.isISOControl(characters[i])) {
```

Comment:
Certain control characters should likely be excluded from this check, e.g.: CR, LF (and I remember seeing another one excluded in one of Jean's PRs - but can't remember which one).